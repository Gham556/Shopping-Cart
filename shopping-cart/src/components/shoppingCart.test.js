import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import ShoppingCart from './shoppingCart';

describe("App Component", () => {
    it("renders correct counter", () => {
        const { container } = render (<ShoppingCart />); 
        expect(container).toMatchSnapshot();
    });

    it("renders 10 after each item is increased by one", () => {
        render(<ShoppingCart />); 
        const button = screen.getAllByRole("button");

        for (let i of button) {
            userEvent.click(i);
        }

        expect(screen.getByTestId('x').textContent).toMatch(/Items In Cart: 10/i)
    })
})