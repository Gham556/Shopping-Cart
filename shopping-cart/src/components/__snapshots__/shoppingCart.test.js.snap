// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders correct counter 1`] = `
<div>
  <div>
    <div
      id="cartHeader"
    >
      <h1>
        Shopping Cart
      </h1>
      <div
        data-testid="x"
      >
        Items In Cart: 
        0
      </div>
    </div>
    <div>
      <div
        class="item"
      >
        <img
          alt=""
          src="stick.png"
        />
        <div
          class="itemDetails"
        >
          <h1
            class="itemName"
          >
            Wooden Sword
          </h1>
          <h3
            class="itemDescription"
          >
            A swift and nimble weapon crafted by Mother Nature Herself
          </h3>
          <div
            class="itemPrice"
          >
            19.99
          </div>
        </div>
        <div
          class="cartAdd"
        >
          <form>
            <label>
              Quantity
            </label>
            <input
              max="10"
              min="0"
              type="number"
            />
            <button
              type="submit"
            >
              Add to Cart
            </button>
          </form>
        </div>
      </div>
      <div
        class="item"
      >
        <img
          alt=""
          src="bucket.jpg"
        />
        <div
          class="itemDetails"
        >
          <h1
            class="itemName"
          >
            Tin Helm
          </h1>
          <h3
            class="itemDescription"
          >
            A sturdy an lightweight helm that can double as a bucket
          </h3>
          <div
            class="itemPrice"
          >
            10.99
          </div>
        </div>
        <div
          class="cartAdd"
        >
          <form>
            <label>
              Quantity
            </label>
            <input
              max="10"
              min="0"
              type="number"
            />
            <button
              type="submit"
            >
              Add to Cart
            </button>
          </form>
        </div>
      </div>
      <div
        class="item"
      >
        <img
          alt=""
          src="car.jpg"
        />
        <div
          class="itemDetails"
        >
          <h1
            class="itemName"
          >
            Chariot
          </h1>
          <h3
            class="itemDescription"
          >
            A perfect travais of yore. (Horses not include)
          </h3>
          <div
            class="itemPrice"
          >
            100
          </div>
        </div>
        <div
          class="cartAdd"
        >
          <form>
            <label>
              Quantity
            </label>
            <input
              max="10"
              min="0"
              type="number"
            />
            <button
              type="submit"
            >
              Add to Cart
            </button>
          </form>
        </div>
      </div>
      <div
        class="item"
      >
        <img
          alt=""
          src="food.jpg"
        />
        <div
          class="itemDetails"
        >
          <h1
            class="itemName"
          >
            Rations
          </h1>
          <h3
            class="itemDescription"
          >
            Healthy and full of fiber. Gaurenteed not to give you scurvy!
          </h3>
          <div
            class="itemPrice"
          >
            5.99
          </div>
        </div>
        <div
          class="cartAdd"
        >
          <form>
            <label>
              Quantity
            </label>
            <input
              max="10"
              min="0"
              type="number"
            />
            <button
              type="submit"
            >
              Add to Cart
            </button>
          </form>
        </div>
      </div>
      <div
        class="item"
      >
        <img
          alt=""
          src="horse.jpeg"
        />
        <div
          class="itemDetails"
        >
          <h1
            class="itemName"
          >
            Horse
          </h1>
          <h3
            class="itemDescription"
          >
            Look no further for your trusty steed. You'll be going places in no-time!
          </h3>
          <div
            class="itemPrice"
          >
            501.99
          </div>
        </div>
        <div
          class="cartAdd"
        >
          <form>
            <label>
              Quantity
            </label>
            <input
              max="10"
              min="0"
              type="number"
            />
            <button
              type="submit"
            >
              Add to Cart
            </button>
          </form>
        </div>
      </div>
      <div
        class="item"
      >
        <img
          alt=""
          src="warhammer.jpeg"
        />
        <div
          class="itemDetails"
        >
          <h1
            class="itemName"
          >
            Warhammer
          </h1>
          <h3
            class="itemDescription"
          >
            Crush your oponents heads like Galagher crushes watermellons!
          </h3>
          <div
            class="itemPrice"
          >
            50.99
          </div>
        </div>
        <div
          class="cartAdd"
        >
          <form>
            <label>
              Quantity
            </label>
            <input
              max="10"
              min="0"
              type="number"
            />
            <button
              type="submit"
            >
              Add to Cart
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
